clmm/gcdata.py

```diff
@@ -221,6 +239,30 @@ def update_cosmo(self, cosmo, overwrite=False):
         """
         self.update_cosmo_ext_valid(self, cosmo, overwrite=overwrite)
 
+    def update_coordinate_system(self, coordinate_system, *args):
+        r"""Updates coordinate_system metadata and converts ellipticity
+
+        Parameters
+        ----------
+        coordinate_system : str
+            Coordinate system of the ellipticity components. Must be either 'celestial' or
+            euclidean'. See https://doi.org/10.48550/arXiv.1407.7676 section 5.1 for more details.
+        *args : tuple
+            Components to be converted, must be in data.
+
+        Returns
+        -------
+        None
+        """
+        in_coordinate_system = self.meta["coordinate_system"]
+        self.update_info_ext_valid("coordinate_system", self, coordinate_system, overwrite=True)
+        if coordinate_system != in_coordinate_system:
```

Comment:
This is where having an else statement with a warning saying the requested coordinate system is already the one of the catalog and therefore no change is being made to the catalog.